kornia/geometry/liegroup/se2.py

```diff
@@ -0,0 +1,207 @@
+# kornia.geometry.so2 module inspired by Sophus-sympy.
+# https://github.com/strasdat/Sophus/blob/master/sympy/sophus/se2.py
+from typing import Optional
+
+from kornia.core import Module, Parameter, Tensor, concatenate, pad, stack, tensor
+from kornia.geometry.liegroup._utils import check_se2_t_shape, check_v_shape
+from kornia.geometry.liegroup.so2 import So2
+from kornia.testing import KORNIA_CHECK_TYPE
+
+
+class Se2(Module):
+    r"""Base class to represent the Se2 group.
+
+    The SE(2) is the group of rigid body transformations about the origin of two-dimensional Euclidean
+    space :math:`R^2` under the operation of composition.
+    See more:
+
+    Example:
+        >>> so2 = So2.identity()
+        >>> t = torch.ones((1, 2))
+        >>> se2 = Se2(so2, t)
+        >>> se2
+        rotation: (1+0j)
+        translation: Parameter containing:
+        tensor([[1., 1.]], requires_grad=True)
+    """
+
+    def __init__(self, r: So2, t: Tensor) -> None:
+        """Constructor for the base class.
+
+        Internally represented by a complex number `z` and a translation 2-vector.
+
+        Args:
+            r: So2 group encompassing a rotation.
+            t: translation vector with the shape of :math:`(B, 2)`.
+
+        Example:
+            >>> so2 = So2.identity()
+            >>> t = torch.ones((1, 2))
+            >>> se2 = Se2(so2, t)
+            >>> se2
+            rotation: (1+0j)
+            translation: Parameter containing:
+            tensor([[1., 1.]], requires_grad=True)
+        """
+        super().__init__()
+        KORNIA_CHECK_TYPE(r, So2)
+        # TODO change to KORNIA_CHECK_SHAPE once there is multiple shape support
+        check_se2_t_shape(t)
+        self._r = r
+        self._t = Parameter(t)
+
+    def __repr__(self) -> str:
+        return f"rotation: {self.r}\ntranslation: {self.t}"
+
+    def __getitem__(self, idx) -> 'Se2':
+        return Se2(self._r[idx], self._t[idx][None])
```

Comment:
```suggestion
        return Se2(self._r[idx], self._t[idx])
```
Maybe we should raise an error if it’s not batched or the index is out of range